From 3fcfbcfb2c1599c1288d41f2b7adecd48236d0ec Mon Sep 17 00:00:00 2001
From: Haibo Chen <haibo.chen@nxp.com>
Date: Tue, 8 Nov 2016 15:06:58 +0800
Subject: [PATCH 1256/5242] MLK-13442: mmc: sdhci-esdhc-imx: release bus
 frequency when usdhc remove

commit  0431ad376dc50e0e3c48a2d6bca1b9535ffd1325 from
https://source.codeaurora.org/external/imx/linux-imx.git

When usdhc driver remove, also need to release bus frequency.

Signed-off-by: Haibo Chen <haibo.chen@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mmc/host/sdhci-esdhc-imx.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/mmc/host/sdhci-esdhc-imx.c b/drivers/mmc/host/sdhci-esdhc-imx.c
index 088dc85..c4ae726 100644
--- a/drivers/mmc/host/sdhci-esdhc-imx.c
+++ b/drivers/mmc/host/sdhci-esdhc-imx.c
@@ -1456,6 +1456,8 @@ static int sdhci_esdhc_imx_remove(struct platform_device *pdev)
 	clk_disable_unprepare(imx_data->clk_per);
 	clk_disable_unprepare(imx_data->clk_ipg);
 	clk_disable_unprepare(imx_data->clk_ahb);
+	if (imx_data->socdata->flags & ESDHC_FLAG_BUSFREQ)
+		release_bus_freq(BUS_FREQ_HIGH);
 
 	sdhci_pltfm_free(pdev);
 
-- 
1.7.9.5

